src/speculators/data_generation/configs.py

```diff
@@ -0,0 +1,140 @@
+"""Configuration registries for data generation pipeline."""
+
+from collections.abc import Callable
+from dataclasses import dataclass
+
+__all__ = [
+    "CHAT_TEMPLATES",
+    "DATASET_CONFIGS",
+    "ChatTemplate",
+    "DatasetConfig",
+    "format_conversation",
+]
+
+
+@dataclass
+class ChatTemplate:
+    """Chat template configuration for a specific model format"""
+
+    name: str
+    system_prompt: str
+    user_header: str
+    assistant_header: str
+    end_of_turn_token: str
+    bos_token: str | None = None
+    eos_token: str | None = None
+
+
+CHAT_TEMPLATES: dict[str, ChatTemplate] = {
```

Comment:
As mentioned in previous comment, I think we will want to eventually use the model's chat template directly, instead of this dataclass + fixed format.